gdx/src/com/badlogic/gdx/graphics/g3d/utils/DefaultTextureBinder.java

```diff
@@ -66,8 +70,13 @@ public DefaultTextureBinder (final int method, final int offset, final int count
 	}
 
 	public DefaultTextureBinder (final int method, final int offset, int count, final int reuseWeight) {
-		final int max = Math.min(getMaxTextureUnits(), MAX_GLES_UNITS);
-		if (count < 0) count = max - offset;
+		int max = Math.min(getMaxTextureUnits(), MAX_GLES_UNITS);
+		if (count < 0) {
+			if (enableMacOSWorkaround && UIUtils.isMac && Gdx.app.getType() == ApplicationType.Desktop)

```

Comment:
There is no other method to know if the application run on macOs? It include a class from scene2d

Reply:
Not as far as I could see. At least none compatible with gwt.

Reply:
@PokeMMO 
`UIUutils#isMac` does not check if it's mac os and is not compatible with GWT, why?
I think it does, also in GWT.
So if the check isMac works, i'd not recommend to add  `DefaultTextureBinder#enableMacOSWorkaround`

btw: #5910 extends UIUtils by `UIUtils#isAndroid` and `UIUtils#isIos`, the pr is currently in progress to test, may if you request changes to this constants, i could look for a solution that ensures that it's mac, if it's not doing right now.

Reply:
`UIUutils#isMac` works fine, we also check if it's a desktop app to avoid interfering with GWT.

`DefaultTextureBinder#enableMacOSWorkaround` exists to allow users to disable this workaround in their own apps more easily.

Reply:
> DefaultTextureBinder#enableMacOSWorkaround exists to allow users to disable this workaround in their own apps more easily.

OK, so my 2nd question: If we have a `DefaultTextureBinder#enableMacOSWorkaround` flag that is only used for some few or only(?) an architecture of mac, why it does not work for all? And how to handle this while deploying, shouldn't do this libgdx automatically, depending on the cpu / system architecture you are using / the app is deployed?

and my third question is:
https://stackoverflow.com/a/42320505 is the answer of you first link, and is the solution, this guy used more than MAX_GLES_UNITS as count, so isn't the solution to reduce the count to the correct value instead of to 1?